### ISSUE TITLE
Loss starts in 0.00000 when training stereo model on custom dataset

### ISSUE THREAD

**Issue by mikeletxeberria** [nianticlabs/monodepth2#271]
Hi,

I am trying to train the stereo model (640x192) on a custom dataset, but the initial loss starts on 0.00000 and does not improve. I have trained the model with KITTI dataset correctly, so something must be wrong in our dataset, or in the configuration I am using. Furthermore, I have correctly trained other mono models without problems. For that, I created a dataloader class (based on KITTI dataloader) and created a new split for our custom dataset. The dataset is composed of 35.000 stereo images. 

The command I use is the following one:

```
python train.py --model_name stereo_640x192 --dataset own --split own --data_path D:\ --png --log_dir D:\mikel\Proyectos\monodepth2\tmp\ --use_stereo --load_weights_folder D:\mikel\Proyectos\monodepth2\models\stereo_640x192 --frame_ids 0 --models_to_load depth encoder
loading model from folder D:\mikel\Proyectos\monodepth2\models\stereo_640x192
Loading depth weights...
Loading encoder weights...
Loading Adam weights
Training model named:
   stereo_640x192
Models and tensorboard events files are saved to:
   D:\mikel\Proyectos\monodepth2\tmp\
Training is using:
   cuda
Training dataset charged.
Validation dataset charged.
Using split:
   own
There are 30.000 training items and 2.500 validation items

epoch   0 | batch      0 | examples/s:   3.1 | loss: 0.00000 | time elapsed: 00h00m36s | time left: 00h00m00s
epoch   0 | batch    250 | examples/s:  40.3 | loss: 0.00000 | time elapsed: 00h02m24s | time left: 22h09m15s
...
```

Here you have two pairs of frames:
Frame 1 left
![001115](https://user-images.githubusercontent.com/49194058/98910742-da9ee080-24c3-11eb-83e9-4cc1c05dc3ff.png)
Frame 1 right
![001115](https://user-images.githubusercontent.com/49194058/98910769-e5f20c00-24c3-11eb-917b-20b595c01cd8.png)
Frame 2 left
![001116](https://user-images.githubusercontent.com/49194058/98912317-35393c00-24c6-11eb-931b-4397a889ab75.png)
Frame 2 right
![001115](https://user-images.githubusercontent.com/49194058/98912347-408c6780-24c6-11eb-9fd9-e4868cc4e269.png)

I have also tryed to train the model with another own stereo dataset that does not have that kind of occlusion and the result is the same. Here you have frames from the second stereo dataset:

![000057](https://user-images.githubusercontent.com/49194058/98912190-0ae77e80-24c6-11eb-80a9-96c7440c3d86.png)
![000057](https://user-images.githubusercontent.com/49194058/98912217-133fb980-24c6-11eb-88cb-f6dcbb6a32a0.png)

Maybe it is the occlusion on the right frame? Or what can be wrong in the configuration? Could you give me some advice?

Thank you,

**Comment by mrharicot** [comment]
Hi,

Are you sure you are loading the images properly in the dataloader? The only reason why you would have a zero loss is if you feed the same image twice. (say left and left).
Moreover it seems that you flipped left and right images in your examples.
Finally, but this is unlikely to be the reason why you have a zero loss, the quality of your images is too low and shows heavy compression artefacts, I would recommend getting much better quality images.

Hope this helps!

**Comment by mikeletxeberria** [comment]
I revised the dataloader class and I had something wrong, now it loads images correctly and the loss looks normal.

Thank you for the help!
